streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -667,8 +668,15 @@ public boolean isProcessable(final long wallClockTime) {
             // thus, the task is not processable, even if there is available data in the record queue
             return false;
         }
-
-        return partitionGroup.readyToProcess(wallClockTime);
+        final boolean readyToProcess = partitionGroup.readyToProcess(wallClockTime);
+        if (!readyToProcess) {
+            if (!timeCurrentIdlingStarted.isPresent()) {
+                timeCurrentIdlingStarted = Optional.of(wallClockTime);
+            }
+        } else {
+            timeCurrentIdlingStarted = Optional.empty();
```

Comment:
Just want to make sure I understand, previously we only considered a task as idling if it was suspended so we're just fixing it up to track the actual idling. And while since KIP-429 suspension is just a transient state that the task passes through right before being closed, it's still used during an upgrade from EAGER. So we're going to keep considering suspension as idling until we can finally drop support for EAGER -- does that sound right?

Reply:
Yes that sounds like what I was thinking